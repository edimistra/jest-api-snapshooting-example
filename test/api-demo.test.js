const { expect, test } = require('@jest/globals');
const axios = require('axios');
const { server, close } = require('./src/server');

function sanitize(data, keys) {
  return keys.reduce((result, key) => {
    const val = _.get(result, key);
    if (!val || _.isArray(val) || _.isObject(val)) {
      return result;
    } else {
      return _.set(_.cloneDeep(result), key, '[SANITIZED]');
    }
  }, data);
}

describe('Run all API tests', () => {
  beforeAll(async () => {
    await server();
  });

  afterAll(() => {
    close();
  });

  test('Get all Ads', (done) => {
    axios.get('http://localhost:3001').then(
      (response) => {
        expect(response).toMatchSnapshot();
        done();
      },
      (error) => {
        done(error);
      },
    );
  });

  test('Insert a new Ad', async () => {
    const response = await axios.post('http://localhost:3001', {
      title: 'Pizza',
      price: 10.5,
    });
    expect(response).toMatchSnapshot();
  });
});
