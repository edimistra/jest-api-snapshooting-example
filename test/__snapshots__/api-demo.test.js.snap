// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run all API tests Get all Ads 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "data": undefined,
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "get",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "http://localhost:3001",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Array [
    Object {
      "_id": "[SANITIZED]",
      "price": 1.3,
      "title": "Soap",
    },
  ],
  "headers": Object {
    "content-length": "63",
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;

exports[`Run all API tests Insert a new Ad 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "data": "{\\"title\\":\\"Pizza\\",\\"price\\":10.5}",
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
      "Content-Type": "application/json;charset=utf-8",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "post",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "http://localhost:3001",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "ad": Object {
      "connection": Object {
        "host": "127.0.0.1",
        "id": 0,
        "port": "[SANITIZED]",
      },
      "insertedCount": 1,
      "insertedId": "[SANITIZED]",
      "n": 1,
      "ok": 1,
      "ops": Array [
        Object {
          "_id": "[SANITIZED]",
          "price": 10.5,
          "title": "Pizza",
        },
      ],
      "result": Object {
        "n": 1,
        "ok": 1,
      },
    },
    "message": "New ad inserted.",
  },
  "headers": Object {
    "content-length": "258",
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;
